test/system/src/test/java/io/pravega/test/system/MultiReaderWriterWithFailOverTest.java

```diff
@@ -0,0 +1,387 @@
+        log.info("Scope passed to readergroup manager {}", scope);
+        readerGroupManager = ReaderGroupManager.withScope(scope, controllerUri);
+        readerGroupManager.createReaderGroup(readerGroupName, ReaderGroupConfig.builder().startingTime(0).build(),
+                Collections.singleton(STREAM_NAME));
+        log.info("Reader group name {} ", readerGroupManager.getReaderGroup(readerGroupName).getGroupName());
+        log.info("Reader group scope {}", readerGroupManager.getReaderGroup(readerGroupName).getScope());
+        log.info("Online readers {}", readerGroupManager.getReaderGroup(readerGroupName).getOnlineReaders());
+
+        //create readers
+        log.info("Creating {} readers", NUM_READERS);
+        List<EventStreamReader<Long>> readerList = new ArrayList<>();
+        String readerName = "reader" + new Random().nextInt(Integer.MAX_VALUE);
+        log.info("Scope that is seen by readers {}", scope);
+        //start reading events
+        for (int i = 0; i < NUM_READERS; i++) {
+            log.info("Starting reader{}", i);
+            log.info("Creating reader with id {}", readerName + i);
+            final EventStreamReader<Long> reader = clientFactory.createReader(readerName + i,
+                    readerGroupName,
+                    new JavaSerializer<Long>(),
+                    ReaderConfig.builder().build());
+            readerList.add(reader);
+        }
+
+        log.info("Test with 2 controller, SSS instances running and without a failover scenario");
+        readWriteTest(writerList, readerList);
+        log.debug("{} with 2 controller, 2 SSS instances succeeds", "multiReaderWriterTest");
+
+        //Scale down SSS instances to 2
+        segmentStoreInstance.scaleService(2, true);
+        Thread.sleep(60000);
+        log.info("Scaling down SSS instances from 3 to 2");
+
+        eventsReadFromPravega = new ConcurrentLinkedQueue<>();
+        stopReadFlag.set(false);
+        eventData = new AtomicLong(); //data used by each of the writers.
+        eventReadCount = new AtomicLong(); // used by readers to maintain a count of events.
+
+        log.info("Test with SSS failover");
+        readWriteTest(writerList, readerList);
+        log.debug("{} with SSS failover succeeds", "multiReaderWriterTest");
+
+        //Scale down controller instances to 2
+        controllerInstance.scaleService(2, true);
+        log.info("Scaling down controller instances from 3 to 2");
+        List<URI> conURIs = controllerInstance.getServiceDetails();
+        // Fetch all the RPC endpoints and construct the client URIs.
+        final List<String> uris = conURIs.stream().filter(uri -> uri.getPort() == 9092).map(URI::getAuthority)
+                .collect(Collectors.toList());
+        controllerURIDirect = URI.create("tcp://" + String.join(",", uris));
+        log.info("Controller Service direct URI: {}", controllerURIDirect);
+        Thread.sleep(60000);
```

Comment:
controllerInstance.scaleService(2, true); here scaleService is a blocking call i.e. it will wait until there are two instances of controller service. Hence, `Thread.sleep()` is not required if the intent is to wait for marathon scale process to finish.